Adds command `spo file roleassignment add`, closes #3702

MathijsVerbeeck opened the pull request:
This PR adds the command `spo file roleassignment add`.

Closes: #3702

milanholemans commented:
Thank you @MathijsVerbeeck we'll review it ASAP!

Adam-it commented on docs/docs/cmd/spo/file/file-roleassignment-add.md:
```diff
@@ -0,0 +1,57 @@
+
+## Examples
+
+Add role assignment to file with id _b2307a39-e878-458b-bc90-03bc578531d6_ in site _https://contoso.sharepoint.com/sites/project-x_ for a principal with id _11_ and role definition id _1073741829_.
```
let's try to avoid duplicating the options value in describing the example 👍

Adam-it commented on src/m365/spo/commands/file/file-roleassignment-add.ts:
```diff
@@ -0,0 +1,270 @@
+    try {
+      const fileUrl: string = await this.getFileURL(args);
+      args.options.roleDefinitionId = await this.getRoleDefinitionId(args.options);
+      if (args.options.upn) {
+        args.options.principalId = await this.getUserPrincipalId(args.options);
+        await this.addRoleAssignment(fileUrl, args.options, logger);
+      }
+      else if (args.options.groupName) {
+        args.options.principalId = await this.getGroupPrincipalId(args.options);
+        await this.addRoleAssignment(fileUrl, args.options, logger);
```
it's a mistake also I did so sorry if you copied/pasted it (I think we should open an issue to correct it in existing commands 🤔). 
Generally, we should not overwrite `args` and consider them as read only. let's create a local variable to keep the `principalId`

Adam-it commented on src/m365/spo/commands/file/file-roleassignment-add.ts:
```diff
@@ -0,0 +1,270 @@
+      else {
+        await this.addRoleAssignment(fileUrl, args.options, logger);
+      }
+
```
let's remove this empty line

Adam-it reviewed (changes requested):
@MathijsVerbeeck awesome work 👍 Your rock 🤩. just small fixes and we are good to go 👍

MathijsVerbeeck commented:
Hi @Adam-it, Thanks for the feedback

I've adjusted the code so that it will use variables instead of modifying the args.options object. Also I've rewritten some of the tests as they weren't yet running async. 

Finally I've rewritten the documentation a bit to become compliant with suggestions that you've on another PR (`spo folder roleinheritance break`)

Adam-it commented:
@MathijsVerbeeck thanks for your awesome work 👍. I will re-review it ASAP. 
In the meantime @pnp/cli-for-microsoft-365-maintainers if anyone would like to have a additional check be my guest 👍

Adam-it commented:
merged manually
